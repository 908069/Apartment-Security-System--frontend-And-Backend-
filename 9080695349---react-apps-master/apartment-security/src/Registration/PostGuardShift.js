import React ,{useState}from 'react'
import axios from 'axios'
function PostGuardShift(props) {
    const [guardShiftDetails,setGuardShiftDetails] = useState({
        guardDate :"",
        guardShift:""

    })
    const handleGuardShift=(e)=>{
        const name =e.target.name
       const value=e.target.value
       setGuardShiftDetails({...guardShiftDetails,[name] : value})
    }
    const handleSubmit=(e)=>{
        e.preventDefault()
        const newRecord={...guardShiftDetails}
        console.log(newRecord)

        axios.post('http://localhost:8082/springfox/api/guard/guardShift',guardShiftDetails)
        .then(response=>{
           console.log(response)
           if(response.status == 200){
           document.getElementById("showSuccessResponseGuardShiftDetails").innerHTML = "Your data entered successfully."
           alert("SignUp Successfully")
           props.history.replace('/login')
        }
           else{
              document.getElementById("showSuccessResponseGuardShiftDetails").innerHTML = "An error occurred while uploading the data."
           }
        })
        .catch(error=>{
           console.log(error)
       })
    }
    return (
        <div>
            <h3>Guard Shift Details</h3>
             <form action ="" onSubmit={handleSubmit}>
            <label htmlFor="guardDate" 
                    >Guard Joining Date</label>
                    <br></br>
                    <input type="text" autoComplete="off" onChange={handleGuardShift}
                     value={guardShiftDetails.guardDate} 
                     name="guardDate" id="guardDate" required/>
                     <br></br>
                     <label htmlFor="guardShift" 
                    >Shift Timing</label>
                    <br></br>
                    <input type="text" autoComplete="off" onChange={handleGuardShift}
                     value={guardShiftDetails.guardShift} 
                     name="guardShift" id="guardShift" required/>
                    <br></br>
                    <button type='submit'>Submit</button>
                </form>
                <div id="showSuccessResponseGuardShiftDetails"></div>
        </div>
    )
}

export default PostGuardShift
