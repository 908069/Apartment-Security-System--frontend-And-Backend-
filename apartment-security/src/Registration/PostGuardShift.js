import React ,{useState}from 'react'
import axios from 'axios'
function PostGuardShift(props) {
    const [guardShiftDetails,setGuardShiftDetails] = useState({
        guardDate :"",
        guardShift:""

    })
    const handleGuardShift=(e)=>{
        const name =e.target.name
       const value=e.target.value
       setGuardShiftDetails({...guardShiftDetails,[name] : value})
    }
    const handleSubmit=(e)=>{
        e.preventDefault()
        const newRecord={...guardShiftDetails}
        console.log(newRecord)

        axios.post('http://localhost:8082/springfox/api/guard/guardShift',guardShiftDetails)
        .then(response=>{
           console.log(response)
           if(response.status == 200){
           document.getElementById("showSuccessResponseGuardShiftDetails").innerHTML = "Your data entered successfully."
           alert("SignUp Successfully")
           props.history.replace('/login')
        }
           else{
              document.getElementById("showSuccessResponseGuardShiftDetails").innerHTML = "An error occurred while uploading the data."
           }
        })
        .catch(error=>{
           console.log(error)
       })
    }
    return (
        <div class = "gh">
            <div className="container">
                    
                    <div className="card col-md-6 offset-md-3 offset-md-3">
            <h3>Guard Shift Details</h3>
             <form action ="" onSubmit={handleSubmit}>
             <div className="form-group">
            <label htmlFor="guardDate" 
                    >Guard Joining Date(yyyy-mm-dd)</label>
                    <br></br>
                    <input type="text" autoComplete="off" onChange={handleGuardShift}
                     value={guardShiftDetails.guardDate} 
                     name="guardDate" id="guardDate" required/>
                     <br></br>
                     <label htmlFor="guardShift" 
                    >Shift Timing</label>
                    <br></br>
                    <input type="text" autoComplete="off" onChange={handleGuardShift}
                     value={guardShiftDetails.guardShift} 
                     name="guardShift" id="guardShift" required/>
                    <br></br>
                    <button class="btn btn-outline-success my-2 my-sm-0" type='submit'>Submit</button>
                    </div>
                </form>
                <div id="showSuccessResponseGuardShiftDetails"></div>
                </div></div>
        </div>
    )
}

export default PostGuardShift
