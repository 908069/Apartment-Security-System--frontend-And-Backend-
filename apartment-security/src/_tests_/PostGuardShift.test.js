import React from 'react';
import PostGuardShift from '../Registration/PostGuardShift';
import renderer from 'react-test-renderer';
 
it('renders correctly', () => {
 const tree = renderer
   .create(<PostGuardShift/>)
   .toJSON();
   console.log(tree)
   expect(tree).toMatchSnapshot();
});