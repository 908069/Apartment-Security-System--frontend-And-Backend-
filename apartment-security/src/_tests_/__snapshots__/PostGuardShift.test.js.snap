// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="gh"
>
  <div
    className="container"
  >
    <div
      className="card col-md-6 offset-md-3 offset-md-3"
    >
      <h3>
        Guard Shift Details
      </h3>
      <form
        action=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="guardDate"
          >
            Guard Joining Date(yyyy-mm-dd)
          </label>
          <br />
          <input
            autoComplete="off"
            id="guardDate"
            name="guardDate"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <br />
          <label
            htmlFor="guardShift"
          >
            Shift Timing
          </label>
          <br />
          <input
            autoComplete="off"
            id="guardShift"
            name="guardShift"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <br />
          <button
            class="btn btn-outline-success my-2 my-sm-0"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
      <div
        id="showSuccessResponseGuardShiftDetails"
      />
    </div>
  </div>
</div>
`;
